src/sst/core/configGraph.cc

```diff
@@ -381,6 +404,64 @@ ConfigComponent* ConfigComponent::findSubComponentByName(const std::string& name
     return nullptr;
 }
 
+ConfigStatistic* ConfigComponent::addStatistic(StatisticId_t sid, const std::string& statisticName)
+{
+    // Check for Enable All Statistics
+    if (statisticName == STATALLFLAG) {
+        // Force the STATALLFLAG to always be on the bottom of the list.
+        // First check to see if anything is in the vector, if vector is empty,
+        // a STATALLFLAG flag will be added to the vector
+        if (false == enabledStatistics.empty()) {
+            // The vector is populated, so see if the STATALLFLAG
+            // already exists if it does, we are done
+            if (STATALLFLAG != enabledStatistics.back().name) {
+                // Add a STATALLFLAG to end of the vector
+                enabledStatistics.emplace_back(STATALLFLAG);
+                return &(enabledStatistics.back());
+            }
+        } else {
+            // Add a STATALLFLAG to end of the vector
+            enabledStatistics.emplace_back(STATALLFLAG);
+            return &(enabledStatistics.back());
+        }
+    } else {
+
+      /* Check for existing statistic with this name */
+      for ( auto &i : enabledStatistics ) {
+          if ( i.name == statisticName)
+          {
+              return nullptr;
+          }
+      }
+
+      enabledStatistics.emplace(enabledStatistics.begin(),
+          ConfigStatistic(sid, id, statisticName));
+
+
+      return &(enabledStatistics.front());
+    }
+
+    return nullptr;
+}
+
+ConfigStatistic* ConfigComponent::findStatistic(StatisticId_t sid)
+{
+    return const_cast<ConfigStatistic*>(const_cast<const ConfigComponent*>(this)->findStatistic(sid));
+}
+
+const ConfigStatistic* ConfigComponent::findStatistic(StatisticId_t sid) const
```

Comment:
I think we can just return a non-const statistic. Even if your method is const, you can still return non-const objects from the container.

Reply:
I did exactly the same logical that already exists for Component or SubComponent : 
```
// ConfigComponent
 ConfigComponent* findSubComponent(ComponentId_t);
 const ConfigComponent* findSubComponent(ComponentId_t) const;

// ConfigGraph
 ConfigComponent* findComponent(ComponentId_t);
 const ConfigComponent* findComponent(ComponentId_t) const;
```
The implementation is also the same with the const_cast : 
```
ConfigComponent* ConfigGraph::findComponent(ComponentId_t id)
{
    return const_cast<ConfigComponent*>(const_cast<const ConfigGraph*>(this)->findComponent(id));
}
```

The **findStatistic** method returns a pointer of a member variable of the **ConfigComponent** class. Indeed, the method itself doesn't change the the member variable, but provides a pointer  : 

It is not possible to simply removed the const qualifier of ConfigStatistic variable of the loop:
```
 // ConfigStatistic* ConfigComponent::findStatistic(StatisticId_t sid) const
 for ( ConfigStatistic &s : enabledStatistics ) // doesn't compile since the const qualifier of the method is dropped.
...
```

The only way to return a non const object here is to use a **const_cast**
```
 // ConfigStatistic* ConfigComponent::findStatistic(StatisticId_t sid) const
 for ( const ConfigStatistic &s : enabledStatistics ) // this line now compiles.
     ConfigStatistic* res = &s; // doesn't compile, a const is requested by 
     ConfigStatistic* res = const_cast<ConfigStatistic*>(&s); // compiles, but with a const_cast.
...
```

Reply:
👍 modified here : 96555ecf2df86e405e57505ca43a4895865108f7